core/src/serve_repair.rs

```diff
@@ -170,6 +167,95 @@ struct ServeRepairStats {
     err_id_mismatch: usize,
 }
 
+struct RepairBudget {
+    last_update: Instant,
+    epoch_staked_nodes: Option<Arc<HashMap<Pubkey, u64>>>,
+    node_used: HashMap<Pubkey, /*bytes*/ u64>,
+    staked_remaining: u64,
+    unstaked_remaining: u64,
+    staked_request: u64,
+    unstaked_request: u64,
+}
+
+impl RepairBudget {
+    const INTERVAL_MS: u64 = 1_000;
```

Comment:
Since you are updating this code, better use `Duration` instead of `u64`.

Reply:
done